overlord/snapstate/policy/base.go

```diff
@@ -98,8 +98,8 @@ func baseUsedBy(st *state.State, baseName string) ([]string, error) {
 			continue
 		}
 
-		for _, si := range snapst.Sequence {
-			snapInfo, err := snap.ReadInfo(name, si)
+		for _, si := range snapst.Sequence.Revisions {
```

Comment:
do we need a method on SnapSequence that returns directly []*snap.SideInfo perhaps? there are never more than a few so it wouldn't be costly

Reply:
Good suggestion, I've added the method and used here and in snapstate where it made the code more readable.